### DISCUSSION TITLE
Mapping between DAG nodes and associated prims on USDImport

### DISCUSSION THREAD

**Discussion by nikshankarnoble** [Autodesk/maya-usd#2554] [Q&A]
Hi all!

Apologies if this is the wrong place for this question - I wasn't sure where else to ask it (if there's a forum for this please let me know!)

I'm looking to find a way to map prims to their corresponding Maya DAG nodes on a USDImport (importing USD as Maya data, not in a stage), even with the same USD file being imported multiple times and Maya automatically renaming nodes. Ideally something along the lines of a `prim_path` attribute on the DAG nodes.

The import is being run through the `file` MEL command in Python and I need to run some processes on nodes in the DAG based on the kind of their associated prim. 

Does anyone have any ideas? Apologies if this is a stupid question, I'm still fairly new to USD integration with Maya.

Best,
Nikhil

**Comment by santosg87** [comment]
@nikshankarnoble just trying to understand your wokflow better, what is it that you are trying to achieve. I'm wondering if there are any other ways to achieve that result, without the need for the DAG node attributes.
